lib/public/ConfigLexicon/ConfigLexiconEntry.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2024 Maxence Lange <maxence@artificial-owl.com>
+ *
+ * @author Maxence Lange <maxence@artificial-owl.com>
+ *
+ * @license AGPL-3.0 or later
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program. If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCP\ConfigLexicon;
+
+/**
+ * Model that represent config values within an app config lexicon.
+ *
+ * @see IConfigLexicon
+ * @since 30.0.0
+ */
+class ConfigLexiconEntry implements IConfigLexiconEntry {
+	private string $definition = '';
+	private ?string $default = null;
+
+	/**
+	 * @param string $key config key
+	 * @param int $valueType type of config value ({@see self::TYPE_STRING} and others)
```

Comment:
```suggestion
	 * @param self::TYPE_* $valueType type of config value ({@see self::TYPE_STRING} and others)
```
Should work fine: https://github.com/nextcloud/spreed/blob/52580808b097054143eca504d165adfbd3195212/lib/Service/ConsentService.php#L60-L66

Reply:
Hmm. just realize we could use enums :]